### ISSUE TITLE
Eslint is not linting on save (with new setting in json)

### ISSUE THREAD

**Issue by JarrodMFlesch** [microsoft/vscode-eslint#843]
We have been using eslint heavily, but with the new update(2.0.4) our files are no longer being linted.

My output dialog box keeps saying `Cannot find module 'babel-eslint'` and `Cannot find module 'eslint-config-airbnb'` if I attempt to use the default formatter. 

Is there a new way to tell eslint to use my current directory? I have all of my node modules installed and set up, I have uninstalled and reinstalled, removed yarn.lock, etc.

I have also updated my vscode settings.json to use the 

```javascript
"editor.codeActionsOnSave": {
        "source.fixAll": true,
        "source.fixAll.eslint": true,
},
```

Any help would be siiiiiiiiick.

Edit: we are using an `.eslintrc.js` file.

**Comment by AbdelrahmanHafez** [comment]
As a quick fix, you can use an older version (1.9.1) by clicking the gear icon beside ESLint on VSCode extensions.

Meanwhile, I made a video demonstrating the auto fix on save issue https://github.com/microsoft/vscode-eslint/issues/841#issuecomment-566814419 Is this the same problem you're having or a different one?

![Screenshot_260](https://user-images.githubusercontent.com/19984935/71046335-ab131700-2140-11ea-8d3a-21e874b02575.jpg)

**Comment by JarrodMFlesch** [comment]
@AbdelrahmanHafez that will be handy, I will definitely be doing that for the time being. Our issue is different because ours never lints, even after multiple saves.

**Comment by wight554** [comment]
Confirm having this issue, tried toggling all auto save toggles available

**Comment by dbaeumer** [comment]
@JarrodMFlesch @wight554 can you provide me with a GitHub repository I can clone that demos what you are seeing. It works for me in various setups and I need a reproducible case to investigate. Thanks!

**Comment by JarrodMFlesch** [comment]
@dbaeumer for sure! This repo is using eslint and does not seem to be linting properly for us: https://github.com/trouble/react-mouse-info

**Comment by RiqueBR** [comment]
I can confirm the above. Following your example on #841, it's not formatting on save.  

`  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  },
  "eslint.enable": true,
  "eslint.run": "onSave",`

Also tried with `source.fixAll: true`, and it didn't work.

**Comment by georgeboot** [comment]
Setting eslint to run `onType` worked for me.

**Comment by dbaeumer** [comment]
@RiqueBR it is currently not working for `"eslint.run": "onSave"`. It is fixrd in master though.

**Comment by RiqueBR** [comment]
Thanks @dbaeumer!

**Comment by dbaeumer** [comment]
@JarrodMFlesch I tested it added the setting to the workspace settings and here is what I get:

![cast](https://user-images.githubusercontent.com/1931590/71185830-799f6600-227c-11ea-83b0-d8034b983f07.gif)

Any additional info?

**Comment by JarrodMFlesch** [comment]
@dbaeumer weird! I'll give it another go and see if it's something on my end (must be). Curious what version eslint are you testing with above?

**Comment by dbaeumer** [comment]
The one that gets installed when running yarn in that repository.

**Comment by YannMCX** [comment]
I've the same problem, but with vue files and eslint-plugin-vue, my vue html <template> block doesn't be linted on save

VSCode version : 1.47.1 26076a4de974ead31f97692a0d32f90d735645c0 x64
VSCode-ESLint : 2.0.6
ESLint : 6.7.2
ESLint-Vue : 6.0.1

ESLint debug output : https://hastebin.com/yukutetogo.log

**Comment by SirCameron** [comment]
Same issue!

**Comment by dbaeumer** [comment]
@YannMCX for some reason the eslint npm module thinks there is nothing to fix. At the end see

```
2019-12-20T10:26:43.861Z eslint:source-code-fixer No fixes to apply
2019-12-20T10:26:43.861Z eslint:cli-engine Linting complete in: 1141ms
```

Can you or @SirCameron provide me with a GitHub repository I can clone that demos waht you are experiencing. Then I can investigate

**Comment by YannMCX** [comment]
@dbaeumer I wasn't able to reproduce the problem on a new Workspace, however, I realized that the lint on save works on small files, not big files...

In the video below, on the left I edit a 456 lines view file, and on the right I edit a 40 lines file, in which the lint on save works...
Vidéo : https://streamable.com/hfdil

**Comment by JarrodMFlesch** [comment]
@dbaeumer hey so I just tested it with `2.0.11` and it seems to be fixed for us! Cool if I close this out or are you still working on things with others in here?

**Comment by JarrodMFlesch** [comment]
Issue fixed with `2.0.11`

**Comment by dbaeumer** [comment]
@YannMCX as pointed out in the Readme you need to increase the `editor.codeActionsOnSaveTimeout` setting for large files.

I am working with the VS Code team to be able to override this setting when ESLint gets installed. But this needs a new VS Code release which will come early next year.

Closing the issue.
